moveit_ros/moveit_servo/include/moveit_servo/servo.h

```diff
@@ -54,8 +54,9 @@ namespace moveit_servo
 class Servo
 {
 public:
-  Servo(const rclcpp::Node::SharedPtr& node, const ServoParameters::SharedConstPtr& parameters,
-        const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor);
+  Servo(const rclcpp::Node::SharedPtr& node,
+        const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor,
+        std::shared_ptr<const servo::ParamListener>& servo_param_listener);
```

Comment:
I wonder if we should be concerned with thread safety here? B/c at least 2 threads will have access to this ParamListener at once (PoseTracker and ServoCalcs).

@tylerjw does ParamListener happen to be threadsafe?

Reply:
I agree with you about thread safety  :+1:  It seems fine since `servo_calcs.cpp` is the only thread that will be updating at runtime.

Reply:
> So, the shared_ptr is always passed by reference. That means, the reference count isn't incremented

Even though it is passed by reference to `Servo` , this is passsed to `ServoCalcs` and  a copy of the `ParamListener`  is created there as `ServoCalcs` needs to have its own copy to check for updates.
If I am correct, the copy of `shared_ptr` [here](https://github.com/ibrahiminfinite/moveit2/blob/4b1083c8b841624462438818ce2a40733ae4cd08/moveit_ros/moveit_servo/src/servo_calcs.cpp#L79) increments the reference count.

Reply:
I looked into it a bit more and turns out we never really needed the `shared_ptr`.
Changed the order of some of the initializations and that removed the need for a `shared_ptr`.
Now there is only a `unqiue_ptr` passed (`std::move`) through all the classes till it reaches `ServoCalcs` 
All classes except `ServoCalcs` only need a copy of `servo::Params` and not the `ParamListener` itself.

Reply:
that's really great!